lightning-macros/src/lib.rs

```diff
@@ -74,3 +77,175 @@ pub fn maybe_await(expr: TokenStream) -> TokenStream {
 
 	quoted.into()
 }
+
+fn expect_ident(token: &TokenTree, expected_name: Option<&str>) {
+	if let TokenTree::Ident(id) = &token {
+		if let Some(exp) = expected_name {
+			assert_eq!(id.to_string(), exp, "Expected ident {}, got {:?}", exp, token);
+		}
+	} else {
+		panic!("Expected ident {:?}, got {:?}", expected_name, token);
+	}
+}
+
+fn expect_punct(token: &TokenTree, expected: char) {
+	if let TokenTree::Punct(p) = &token {
+		assert_eq!(p.as_char(), expected, "Expected punctuation {}, got {}", expected, p);
+	} else {
+		panic!("Expected punctuation {}, got {:?}", expected, token);
+	}
+}
+
+/// Scans a match statement for fields which should be skipped
+///
+/// Wraps a `match self {..}` statement and scans the fields in the match patterns (in the form
+/// `ref $field_name: $field_ty`) for types marked `legacy`, skipping those fields.
+#[proc_macro]
+pub fn skip_legacy_fields(expr: TokenStream) -> TokenStream {
+	let mut stream = expr.clone().into_iter();
+	let mut res = TokenStream::new();
+
+	let match_ident = stream.next().unwrap();
+	expect_ident(&match_ident, Some("match"));
+	res.extend(proc_macro::TokenStream::from(match_ident));
+
+	let self_ident = stream.next().unwrap();
+	expect_ident(&self_ident, Some("self"));
+	res.extend(proc_macro::TokenStream::from(self_ident));
+
+	let token_to_stream = |tok| proc_macro::TokenStream::from(tok);
+
+	let arms = stream.next().unwrap();
+	if let TokenTree::Group(group) = arms {
+		let mut new_arms = TokenStream::new();
+
+		let mut arm_stream = group.stream().into_iter().peekable();
+		while arm_stream.peek().is_some() {
+			let enum_ident = arm_stream.next().unwrap();
+			let co1 = arm_stream.next().unwrap();
+			expect_punct(&co1, ':');
+			let co2 = arm_stream.next().unwrap();
+			expect_punct(&co2, ':');
```

Comment:
I am just putting it there for reference, if you are looking for a no dependencies parser for  the proc macro I developed a PoC for the rust compiler a while back https://github.com/rsmicro/kproc-macros this simplifies the parsing a little bit IMHO, but not sure if it is worth for just a single proc macro